detectable/src/main/java/com/synopsys/integration/detectable/detectables/clang/linux/LinuxDistro.java

```diff
@@ -17,9 +19,29 @@ public class LinuxDistro {
 
     public Optional<String> extractLinuxDistroNameFromEtcDir() {
         File etcDir = new File("/etc");
+
+        if (isOracleLinux(etcDir)) {
+            logger.trace("Target image Linux distro is an Oracle Linux");
+            return Optional.of("oracle_linux");
+        }
+
         return extractLinuxDistroNameFromEtcDir(etcDir);
     }
 
+    public boolean isOracleLinux(File etcDirFile) {
+        for (File etcFile : etcDirFile.listFiles()) {
```

Comment:
It seems like we call this isOracleLinux method first to check if we are dealing with OracleLinux and if not then we call extractLinuxDistroNameFromEtcDir. My concern is that it seems both do a for loop on the /etc directory to determine what linux distro we are dealing with. What it be possible to push some of this isOracleLinux logic down into extractLinuxDistroNameFromEtcDir? My concern is that in most cases I'm guessing the customer won't have Oracle Linux and we'll end up walking the directory structure twice.

Reply:
We shall discuss it in a separate call.

Reply:
@dterrysynopsys I have described the reason below.